pandas/core/groupby/grouper.py

```diff
@@ -607,11 +607,9 @@ def is_in_axis(key) -> bool:
 
     # if the grouper is obj[name]
     def is_in_obj(gpr) -> bool:
-        if not hasattr(gpr, "name"):
-            return False
         try:
-            return gpr is obj[gpr.name]
-        except (KeyError, IndexError):
+            return id(gpr) == id(obj[gpr.name])
+        except Exception:
```

Comment:
Probably don't want to do this. I think the fix should rather be in `Series.__contains__`.

```python

In [11]: s = pd.Series([1, 2, 3], index=pd.period_range('2000', periods=3, name='A'))

In [12]: "A" in s
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-12-6445c229fb58> in <module>
----> 1 "A" in s

~/Envs/dask-dev/lib/python3.7/site-packages/pandas/core/generic.py in __contains__(self, key)
   1737     def __contains__(self, key) -> bool_t:
   1738         """True if the key is in the info axis"""
-> 1739         return key in self._info_axis
   1740
   1741     @property

~/Envs/dask-dev/lib/python3.7/site-packages/pandas/util/_decorators.py in wrapper(*args, **kwargs)
    353         @wraps(func)
    354         def wrapper(*args, **kwargs) -> Callable:
--> 355             return func(*args, **kwargs)
    356
    357         # collecting docstring and docstring templates

~/Envs/dask-dev/lib/python3.7/site-packages/pandas/core/indexes/period.py in __contains__(self, key)
    332             hash(key)
    333             try:
--> 334                 self.get_loc(key)
    335                 return True
    336             except KeyError:

~/Envs/dask-dev/lib/python3.7/site-packages/pandas/core/indexes/period.py in get_loc(self, key, method, tolerance)
    501
    502             try:
--> 503                 asdt, reso = parse_time_string(key, self.freq)
    504             except DateParseError as err:
    505                 # A string with invalid format

pandas/_libs/tslibs/parsing.pyx in pandas._libs.tslibs.parsing.parse_time_string()

pandas/_libs/tslibs/parsing.pyx in pandas._libs.tslibs.parsing.parse_datetime_string_with_reso()

ValueError: Given date string not likely a datetime.
```

should raise a KeyError?

Reply:
Or perhaps the fix would be in `Series.__getitem__`.

Reply:
the plan is to backport this, so hoping to keep changes minimal. (if not reverting)

added a TODO as suggested https://github.com/pandas-dev/pandas/pull/34049#discussion_r421570058 for now

Reply:
make sense.  longer-term i think Tom's approach is optimal